### PR TITLE
feat: add utils.PollE

### PR DESCRIPTION
**Pull request by grayside**
This is part one of fixing #1726, splitting `utils.Poll` into two functions, one that gives the caller insights to decide how to handle failure, and the other preserving existing behavior.

Additional, this PR adds initial test coverage to Polling behavior. If the iteration count were not loop-scoped, I might have made this `PollE(...) (int, error)` and returned the iteration count as a first value.

### CODE REVIEW

**Comment by grayside** [comment]
Fixed now, ready for review.

File: infra/blueprint-test/pkg/utils/poll.go [hunk 1]
```diff
@@ -45,10 +55,12 @@ func Poll(t testing.TB, condition func() (bool, error), numRetries int, interval
 	}
 
 	if err != nil {
-		t.Fatalf("failed to pull provided condition after %d retries, last error: %v", numRetries, err)
+		return fmt.Errorf("failed to pull provided condition after %d retries, last error: %v", numRetries, err)
```
Comment (bharathkkb) [inline infra/blueprint-test/pkg/utils/poll.go:58]: nit: would custom errors be beneficial for users distinguish between errors types?

**Review by bharathkkb — APPROVED** [review]
Thanks for adding tests too! Looks like golint is complaining about something but LGTM otherwise

File: infra/blueprint-test/pkg/utils/poll.go (hunk 1 above)
Comment (grayside) [reply infra/blueprint-test/pkg/utils/poll.go]: Added.

**Pull request merged**